glsk/glsk-document-cse/src/main/java/com/powsybl/glsk/cse/CseGlskPoint.java

```diff
@@ -35,41 +36,53 @@ public CseGlskPoint(TimeSeriesType timeSeries) {
         BusinessTypeList businessType = timeSeries.getBusinessType().getV();
 
         try {
+            //pretreatment to set sum of participation factors of all blocks to 1 (100%)
+
+            //calculate sum of block factors
+            BigDecimal sumBlockFactors = Stream.concat(Stream.ofNullable(timeSeries.getManualLSKBlockOrPropLSKBlock()),
```

Comment:
Maybe putting all this `Stream` operation in a well-named dedicated method would make the code easier to understand and make the comment above unnecessary.